File: app/src/main/java/br/com/douglaspac/qualcombustivel/SaveSharedPreferences.java
package br.com.douglaspac.qualcombustivel;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

/**
 * Created by douglasmartins on 24/04/17.
 */

public class SaveSharedPreferences
{
    static final String PREF_GAS = "gas";
    static final String PREF_ETANOL = "etanol";

    static SharedPreferences getSharedPreferences(Context ctx)
    {
        return PreferenceManager.getDefaultSharedPreferences(ctx);
    }

    public static void setGas(Context ctx, Float gas)
    {
        SharedPreferences.Editor editor = getSharedPreferences(ctx).edit();
        editor.putFloat(PREF_GAS, gas);
        editor.commit();
    }

    public static Float getGas(Context ctx)
    {
        return getSharedPreferences(ctx).getFloat(PREF_GAS, (float) 3.000);
    }

    public static void setEtanol(Context ctx, Float etanol)
    {
        SharedPreferences.Editor editor = getSharedPreferences(ctx).edit();
        editor.putFloat(PREF_ETANOL, etanol);
        editor.commit();
    }

    public static Float getEtanol(Context ctx)
    {
        return getSharedPreferences(ctx).getFloat(PREF_ETANOL, (float) 2.000);
    }


}

File: app/src/main/java/br/com/douglaspac/qualcombustivel/SaveSharedPreferences.kt
package br.com.douglaspac.qualcombustivel.kotlin

import android.content.Context
import android.content.SharedPreferences
import android.preference.PreferenceManager

class SaveSharedPreferences {
    companion object {
        internal val PREF_GAS = "gas"
        internal val PREF_ETANOL = "etanol"

        private fun getSharedPreferences(ctx: Context) : SharedPreferences {
            return PreferenceManager.getDefaultSharedPreferences(ctx)
        }
        fun setGas(ctx: Context, gas: Float?) {
            val editor = getSharedPreferences(ctx).edit()
            if(gas != null){
                editor.putFloat(PREF_GAS, gas)
                editor.apply()
            }
        }

        fun getGas(ctx: Context): Float {
            return getSharedPreferences(ctx).getFloat(PREF_GAS, 3.000.toFloat())
        }

        fun setEtanol(ctx: Context, etanol: Float?) {
            val editor = getSharedPreferences(ctx).edit()
            if(etanol != null) {
                editor.putFloat(PREF_ETANOL, etanol)
                editor.apply()
            }
        }

        fun getEtanol(ctx: Context): Float {
            return getSharedPreferences(ctx).getFloat(PREF_ETANOL, 2.000.toFloat())
        }
    }
}